### PR TITLE
Update `@web5/credentials` README

### PR DESCRIPTION
**Pull request by nitro-neal**
# Credentials

The Credentials package enables the creation, signing, verification, and general processing of `Verifiable Credentials` (VCs). It also has full `Presentation Exchange` support.

### Features

- Create Verifiable Credentials with flexible data types.
- Sign credentials using decentralized identifiers (DIDs).
- Verify the integrity and authenticity of VCs encoded as JSON Web Tokens (JWTs).
- Parse JWT representations of VCs into `VerifiableCredential` instances.

### Creating a Verifiable Credential

Create a new `VerifiableCredential` with the following parameters:

- `type`: Type of the credential.
- `issuer`: Issuer URI.
- `subject`: Subject URI.
- `data`: Credential data.
- `expirationDate?`: (optinal) Expiration Date

```javascript
class StreetCredibility {
  constructor(localRespect, legit) {
    this.localRespect = localRespect;
    this.legit = legit;
  }
}

const vc = new VerifiableCredential({
  type: "StreetCred",
  issuer: "did:example:issuer",
  subject: "did:example:subject",
  data: new StreetCredibility("high", true)
});
```

### Signing a Verifiable Credential
Sign a `VerifiableCredential` with a DID:

- `did`: The did that is signing the VC

First create a `Did` object as follows:
```javascript
import { DidKeyMethod } from '@web5/dids';
const issuer = await DidKeyMethod.create();
```

Then sign the VC using the `did` object
```javascript
const vcJwt = vc.sign({ did: issuer });
```

### Verifying a Verifiable Credential
Verify the integrity and authenticity of a Verifiable Credential

- `vcJwt`: The VC in JWT format as a String.
```javascript
try {
  await VerifiableCredential.verify({ vcJwt: signedVcJwt })
  console.log("VC Verification successful!")
} catch (e: Error) {
  console.log("VC Verification failed: ${e.message}")
}
```

### Parsing a JWT into a Verifiable Credential
Parse a JWT into a `VerifiableCredential` instance

`vcJwt`: The VC JWT as a String.

```javascript
const vc = VerifiableCredential.parseJwt({ vcJwt: signedVcJwt })
```

## Presentation Exchange

`PresentationExchange` is designed to facilitate the creation of a Verifiable Presentation by providing tools to select and validate Verifiable Credentials against defined criteria.

### Features

- Select credentials that satisfy a given presentation definition.
- Validate if a Verifiable Credential JWT satisfies a Presentation Definition.
- Validate input descriptors within Verifiable Credentials.

### Selecting Credentials
Select Verifiable Credentials that meet the criteria of a given presentation definition.

- `vcJwts`: The list of Verifiable Credentials to select from.
- `presentationDefinition` The Presentation Definition to match against.

This returns a list of the vcJwts that are acceptable in the presentation definition.
```javascript
const selectedCredentials = PresentationExchange.selectCredentials({
    vcJwts: signedVcJwts,
    presentationDefinition: presentationDefinition
})
```

### Satisfying a Presentation Definition
Validate if a Verifiable Credential JWT satisfies the given presentation definition. Will return an error if the evaluation results in warnings or errors.

- `vcJwts`: The list of Verifiable Credentials to select from.
- `presentationDefinition` The Presentation Definition to match against.

```javascript 
try {
  PresentationExchange.satisfiesPresentationDefinition({vcJwts: signedVcJwts, presentationDefinition: presentationDefinition})
  console.log("vcJwts satisfies Presentation Definition!")
} catch (e: Error) {
  console.log("Verification failed: ${e.message}")
}
```

### Create Presentation From Credentials
Creates a presentation from a list of Verifiable Credentials that satisfy a given presentation definition. This function initializes the Presentation Exchange (PEX) process, validates the presentation definition, evaluates the credentials against the definition, and finally constructs the presentation result if the evaluation is successful.

- `vcJwts`: The list of Verifiable Credentials to select from.
- `presentationDefinition` The Presentation Definition to match against.

```javascript
const presentationResult = PresentationExchange.createPresentationFromCredentials({vcJwts: signedVcJwts, presentationDefinition: presentationDefinition})
```

### Validate Definition
This method validates whether an object is usable as a presentation definition or not.

- `presentationDefinition` The Presentation Definition to validate

```javascript
const valid = PresentationExchange.validateDefinition({presentationDefinition})
```

### Validate Submission
This method validates whether an object is usable as a presentation submission or not.

- `presentationSubmission` The Presentation Submission to validate 

```javascript
const valid = PresentationExchange.validateSubmission({presentationSubmission})
```

### Validate Presentation
Evaluates a presentation against a presentation definition.

- `presentationDefinition` The Presentation Definition to validate
- `presentation` The Presentation

```javascript
const evaluationResults = PresentationExchange.evaluatePresentation({presentationDefinition, presentation})
```

### CODE REVIEW

**Comment by mistermoe** [comment]
yo @nitro-neal ! i think we're missing an `await` before `vc.sign` and `VerifiableCredential.create`

**Comment by frankhinek** [comment]
We could also add docs for using the `Jwt` class but this functionality is going to end up moving over to `@web5/crypto` after [this](https://github.com/TBD54566975/web5-js/issues/271) and [this](https://github.com/TBD54566975/web5-js/issues/347) are done.

File: packages/credentials/README.md [hunk 1]
```diff
@@ -111,13 +111,11 @@ Validate if a Verifiable Credential JWT satisfies the given presentation definit
 
 ```javascript 
 try {
-  PresentationExchange.satisfiesPresentationDefinition(vcJwts, presentationDefinition)
+  PresentationExchange.satisfiesPresentationDefinition({vcJwts: signedVcJwts, presentationDefinition: presentationDefinition})
```
Comment (frankhinek) [inline packages/credentials/README.md:114]:
```suggestion
  PresentationExchange.satisfiesPresentationDefinition({ vcJwts: signedVcJwts, presentationDefinition: presentationDefinition })
```

File: packages/credentials/README.md [hunk 2]
```diff
@@ -127,26 +125,33 @@ Creates a presentation from a list of Verifiable Credentials that satisfy a give
 - `presentationDefinition` The Presentation Definition to match against.
 
 ```javascript
-const presentationResult = PresentationExchange.createPresentationFromCredentials(vcJwts, presentationDefinition)
+const presentationResult = PresentationExchange.createPresentationFromCredentials({vcJwts: signedVcJwts, presentationDefinition: presentationDefinition})
```
Comment (frankhinek) [inline packages/credentials/README.md:128]:
```suggestion
const presentationResult = PresentationExchange.createPresentationFromCredentials({ vcJwts: signedVcJwts, presentationDefinition: presentationDefinition })
```

File: packages/credentials/README.md [hunk 3]
```diff
@@ -127,26 +125,33 @@ Creates a presentation from a list of Verifiable Credentials that satisfy a give
...
 ```
 
 ### Validate Definition
 This method validates whether an object is usable as a presentation definition or not.
 
+- `presentationDefinition` The Presentation Definition to validate
+
 ```javascript
-const valid = PresentationExchange.validateDefinition(presentationDefinition)
+const valid = PresentationExchange.validateDefinition({presentationDefinition})
```
Comment (frankhinek) [inline packages/credentials/README.md:137]:
```suggestion
const valid = PresentationExchange.validateDefinition({ presentationDefinition })
```

File: packages/credentials/README.md [hunk 4]
```diff
@@ -127,26 +125,33 @@ Creates a presentation from a list of Verifiable Credentials that satisfy a give
...
 ```
 
 ### Validate Submission
 This method validates whether an object is usable as a presentation submission or not.
 
+- `presentationSubmission` The Presentation Submission to validate 
+
 ```javascript
-const valid = PresentationExchange.validateSubmission(presentationSubmission)
+const valid = PresentationExchange.validateSubmission({presentationSubmission})
```
Comment (frankhinek) [inline packages/credentials/README.md:146]:
```suggestion
const valid = PresentationExchange.validateSubmission({ presentationSubmission })
```

File: packages/credentials/README.md [hunk 5]
```diff
@@ -127,26 +125,33 @@ Creates a presentation from a list of Verifiable Credentials that satisfy a give
...
 
 ### Validate Presentation
 Evaluates a presentation against a presentation definition.
 
+- `presentationDefinition` The Presentation Definition to validate
+- `presentation` The Presentation
+
 ```javascript
-const evaluationResults = PresentationExchange.evaluatePresentation(presentationDefinition, presentation)
+const evaluationResults = PresentationExchange.evaluatePresentation({presentationDefinition, presentation})
```
Comment (frankhinek) [inline packages/credentials/README.md:156]:
```suggestion
const evaluationResults = PresentationExchange.evaluatePresentation({ presentationDefinition, presentation })
```

File: packages/credentials/tests/presentation-exchange.spec.ts [hunk 6]
```diff
@@ -167,7 +167,9 @@ describe('PresentationExchange', () => {
         vcJwts: [btcCredentialJwt],
         presentationDefinition
       });
-      const result = PresentationExchange.validateSubmission(presentationResult.presentationSubmission);
+
+      const presentationSubmission = presentationResult.presentationSubmission;
+      const result = PresentationExchange.validateSubmission({presentationSubmission});
```
Comment (frankhinek) [inline packages/credentials/tests/presentation-exchange.spec.ts:172]:
```suggestion
      const result = PresentationExchange.validateSubmission({ presentationSubmission });
```

File: packages/credentials/tests/presentation-exchange.spec.ts [hunk 7]
```diff
@@ -184,7 +186,8 @@ describe('PresentationExchange', () => {
       expect(presentationEvaluationResults.errors).to.deep.equal([]);
       expect(presentationEvaluationResults.warnings).to.deep.equal([]);
 
-      const result = PresentationExchange.validateSubmission(presentationResult.presentationSubmission);
+      const presentationSubmission = presentationResult.presentationSubmission;
+      const result = PresentationExchange.validateSubmission({presentationSubmission});
```
Comment (frankhinek) [inline packages/credentials/tests/presentation-exchange.spec.ts:190]:
```suggestion
      const result = PresentationExchange.validateSubmission({ presentationSubmission });
```

**Review by frankhinek — CHANGES_REQUESTED** [review]

**Review by frankhinek — APPROVED** [review]

**Review by mistermoe — APPROVED** [review]

**Pull request merged**